topdown/time.go

```diff
@@ -19,6 +19,28 @@ import (
 var tzCache map[string]*time.Location
 var tzCacheMutex *sync.Mutex
 
+// -9223372036854775808ns, int64 wraparound point
+var minDateAllowedForNsConversion = time.Date(1677, 9, 21, 0, 12, 43, 145224192, time.UTC)
+
+// 9223372036854775807ns, int64 wraparound point
+var maxDateAllowedForNsConversion = time.Date(2262, 4, 11, 23, 47, 16, 854775807, time.UTC)
+
+func toSafeUnixNano(t time.Time) (int64, error) {
+	if t.Before(minDateAllowedForNsConversion) {
+		// Earlier dates than this exhibits undefined Time.UnixNano() behaviour.
+		return 0, fmt.Errorf("time is before %v, and cannot be converted to epoch nanoseconds",
+			minDateAllowedForNsConversion)
+	}
+
+	if t.After(maxDateAllowedForNsConversion) {
+		// Later dates than this exhibits undefined Time.UnixNano() behaviour.
+		return 0, fmt.Errorf("time is after %v, and cannot be converted to epoch nanoseconds",
+			maxDateAllowedForNsConversion)
```

Comment:
💭 Hmm this is quite explicit. I'd have gone with "time outside of valid range" or something. But that's just a preference, OK too keep it as-is, I think.

Reply:
Your suggestion is more concise, so I'll go with that.